lib/enumerize/value.rb

```diff
@@ -17,6 +17,13 @@ def text
       I18n.t(i18n_keys[0], :default => i18n_keys[1..-1])
     end
 
+    def ==(other)
+      return super(other) if other.is_a? String
+      return super(other.to_s) if other.is_a? Symbol
+      return value == other if other.is_a?(Integer) && value.is_a?(Integer)
+      super other
```

Comment:
It seems to me that the `==` implementation can be simplified (IMO it has too many `is_a?` checks).

I guess that `super(other.to_s) || value == other` should work.